tensorflow/python/kernel_tests/pooling_ops_test.py

```diff
@@ -1271,11 +1271,8 @@ def _testMaxPoolGradDirectWithNans2_1(self):
       return
 
     # Test the GPU implementation that uses cudnn for now.
-    # It does not propagate the diff in cases of NaNs
-    expected_input_backprop_cudnn = [
-        0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
-        0.0, 0.0
-    ]
+    # It also propagates the diff in cases of NaNs
+    expected_input_backprop_cudnn = expected_input_backprop_tf_cpu
```

Comment:
Both ways need to be tested.

Reply:
Is it sufficient to make the test respect the environment variable? Currently, toggling NaN propagation off and on in a single test is not possible.

Reply:
Please look at my other comment for this. It's better to move the setting into the graph, meaning in the OpKernel constructors.